conans/client/rest/rest_client_v1.py

```diff
@@ -203,31 +203,31 @@ def _get_package_urls(self, package_reference):
 
         return urls
 
-    def upload_recipe(self, conan_reference, the_files, retry, retry_wait, ignore_deleted_file,
-                      no_overwrite):
+    def upload_recipe(self, conan_reference, the_files, retry, retry_wait, no_overwrite):
         """
         the_files: dict with relative_path: content
         """
         self.check_credentials()
 
         # Get the remote snapshot
         remote_snapshot = self._get_conan_snapshot(conan_reference)
-        local_snapshot = {filename: md5sum(abs_path) for filename, abs_path in the_files.items()}
 
-        # Get the diff
-        new, modified, deleted = diff_snapshots(local_snapshot, remote_snapshot)
-        if ignore_deleted_file and ignore_deleted_file in deleted:
-            deleted.remove(ignore_deleted_file)
+        if remote_snapshot and no_overwrite != "force":
+            remote_manifest = self.get_conan_manifest(conan_reference)
+            local_manifest = FileTreeManifest.loads(load(the_files["conanmanifest.txt"]))
 
-        if not new and not deleted and modified in (["conanmanifest.txt"], []):
-            return False, conan_reference
+            if remote_manifest == local_manifest:
+                return False, conan_reference
 
-        if no_overwrite and remote_snapshot:
             if no_overwrite in ("all", "recipe"):
                 raise ConanException("Local recipe is different from the remote recipe. "
                                      "Forbidden overwrite")
+
+        local_snapshot = {filename: None for filename, abs_path in the_files.items()}
         files_to_upload = {filename.replace("\\", "/"): the_files[filename]
-                           for filename in new + modified}
+                           for filename in local_snapshot}
```

Comment:
`files_to_upload = {filename.replace("\\", "/"): abs_path for filename, abs_path in the_files.items()}`

And I'm also wondering if this replacement is needed at all

Reply:
Just, it is a bit ugly, but I prefer to leave it by now, too many changes.